### ISSUE TITLE
Can you please explain order of data?

### ISSUE THREAD

**Issue by NRshka** [HobbitLong/SupContrast#10]
I apologize for the stupid question. I read the original article, but could not connect it with the implementation. I try to run simple code:
```python
    features = torch.randn(256, 2, 128)
    features = F.normalize(features, dim=1)
    labels = torch.randint(10, size=(256,))
    criterion(features, labels)
```
and get nan.
This occurs because some values in exp_logits.sum (1, keepdim = True) are zero. What are the restrictions on the features/labels?

**Comment by dcato98** [comment]
@NRshka You're normalizing along the n_views dimension, but you need to be normalizing along the feature dimension. Features are described by: (`batch_size`, `n_views`, `feature_dim`). `n_views` is the number of augmented versions of an image in the batch - since they originate from the same image, we know they're positives. `feature_dim` is the dimension of the learned embedding.

So, all you need to do is change "dim=1" to "dim=2" and it should work as intended.

I set all relevant seeds to 0 and got a loss of `tensor(7.0303)` using your code above, and replacing criterion with SupConLoss() from 'losses.py'.

**Comment by NRshka** [comment]
I really appreciate your explanation!
